### PR TITLE
widget on new landing page

### PR DESCRIPTION
**Pull request by will-weiss**

### CODE REVIEW

File: src/clients/api.ts [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import algoliasearch from 'algoliasearch'
+import { createIssue } from './mockApi' // TODO - move this function to somewhere outside mockApi
+
+import { issueFromJson } from './util'
+
+// TODO - do something if they are not defined
```
Comment (will-weiss) [inline src/clients/api.ts]: throw an error - that way the build will fail, which is what we want.

File: src/clients/api.ts [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+export async function postIssue(issuePostBody: IssuePostBody): Promise<Issue> {
+  const issue = createIssue(issuePostBody)
+  const issueStr = JSON.stringify(issue)
+
+  // tslint:disable-next-line: no-expression-statement
+  await fetch('/api/postIssue', {
+    method: 'POST',
+    body: issueStr,
+  }) // TODO - do we want to handle any errors here?
```
Comment (will-weiss) [inline src/clients/api.ts]: Let the caller handle the error, this is fine

File: src/clients/api.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  const issues = result.hits.map(hit => {
+    const { objectID, ...rest } = hit
+    return issueFromJson(JSON.stringify({ id: objectID, ...rest })) // TODO - remove access attributes from rest
+  })
+
+  return issues
+}
+
+export async function postIssue(issuePostBody: IssuePostBody): Promise<Issue> {
+  const issue = createIssue(issuePostBody)
```
Comment (will-weiss) [inline src/clients/api.ts:27]: how are we handling "id"?

File: functions/postIssue.js [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+  } catch (error) {
+    return {
+      statusCode: 500,
+      body: 'Failed to execute function. Could not save record in the database',
+    }
+  }
+
+  return {
+    statusCode: 200,
+    body: 'Data was inserted posted successfully',
```
Comment (will-weiss) [inline functions/postIssue.js]: Is there data in the `result` that we should return to the client?

Comment (shachar-langer) [reply functions/postIssue.js]: The client already has the `issues` object. The only thing it needs is a confirmation that the issue was posted. I'm using the `.wait()` method of Algolia's client to make sure the data was saved.

**Pull request merged**